contracts/src/structs/Subnet.sol

```diff
@@ -153,6 +154,17 @@ struct GenesisValidator {
     bytes metadata;
 }
 
+/// @notice Maintains the genesis information related to a particular subnet
+struct SubnetGenesis {
+    /// The genesis validators
+    EnumerableSet.AddressSet validators;
+    /// Tracks the validator info. This is only populated when the subnet is bootstrapped
+    mapping(address => GenesisValidator) validatorInfo;
+
+    /// TODO: migrating all the genesis related fields to this struct so that one can handle
+    ///       them all in a library.
```

Comment:
Does it make sense to do this now, to minimise the migration logic we'll need to write to upgrade early adopters?